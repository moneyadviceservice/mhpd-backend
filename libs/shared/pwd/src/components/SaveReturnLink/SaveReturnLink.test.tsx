import React from 'react';
import { render, screen } from '@testing-library/react';
import { SaveReturnLink } from '.';

jest.mock('next/router', () => ({
  useRouter: jest.fn().mockReturnValue({
    query: {},
  }),
}));

describe('Save and return component', () => {
  it('renders correctly', () => {
    render(
      <SaveReturnLink
        href={{
          pathname: '/path-to-save-page',
          query: 'language=en',
        }}
        testId="save-return-test"
      />,
    );
    const saveAndReturn = screen.getByTestId('save-return-test');
    expect(saveAndReturn).toMatchSnapshot();
  });
});
