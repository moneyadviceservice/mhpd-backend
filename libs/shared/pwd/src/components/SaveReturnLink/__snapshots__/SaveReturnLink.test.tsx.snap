// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save and return component renders correctly 1`] = `
<a
  class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
  data-testid="save-return-test"
  href="/path-to-save-page?language=en"
>
  <svgr-mock
    classname=""
  />
   
  Save and come back later
</a>
`;
